DocBook. Full-Stack Capstone Project.

CaptainBawa opened the pull request:
### In this project:

- [ ] We completed the following:
     - Built the frontend with react and redux.
     - Add components to perform various actions based on the project requirements.
     - Add responsiveness to the app.
     - Fetched data from the backend api.
     - Deployed the app.

CaptainBawa committed:
Create LICENSE

CaptainBawa committed:
Setup react using vite

CaptainBawa committed:
Setup linters

CaptainBawa committed:
Add react-router-dom and redux toolkit

CaptainBawa committed:
Install axios and react-redux

CaptainBawa committed:
Setup to use redux store

CaptainBawa committed:
Setup to use routes

CaptainBawa committed:
Delete dev

CaptainBawa committed:
Setup redux store

CaptainBawa committed:
Setup doctors slice to fetch data from the backend

CaptainBawa committed:
Setup appointments slice to fetch data from the backend

CaptainBawa committed:
Fix linters error

CaptainBawa committed:
Merge pull request #1 from CaptainBawa/redux-store

DocBook. Milestone: Redux Store

CaptainBawa committed:
Add appointments API link

CaptainBawa committed:
Add doctors API link

Christelle-12 committed:
Add doctors list page

Christelle-12 committed:
Wrapped doctor components in a directory

Christelle-12 committed:
Delete DoctorList component

tomdan-ai committed:
Correct linter errors

tomdan-ai committed:
Correct linter errors

tomdan-ai committed:
Correct JS Linters

tomdan-ai committed:
Configure Vite Live server

tomdan-ai committed:
Create Navigation menu for DocBook app

tomdan-ai committed:
Import and implement Navigation menu in root

tomdan-ai committed:
Style navigation menu

tomdan-ai committed:
Adjust store to avoid linter errors

tomdan-ai committed:
Install and implement tailwind Css in app

tomdan-ai committed:
Adjust Title

Christelle-12 committed:
Add create doctor component

tomdan-ai committed:
Merge pull request #2 from CaptainBawa/Navigation

DocBook: Navigation

Christelle-12 committed:
Merge branch 'development' into doctor-pages

Christelle-12 committed:
Add DoctorsList component

Christelle-12 committed:
Fix linters

ridabensalem committed:
Merge pull request #4 from CaptainBawa/auth

DocBook. Milestone: Login & Sign-in

Christelle-12 committed:
Add Doctorlist styles

Christelle-12 committed:
Fix linters

tomdan-ai committed:
Merge branch 'development' into doctor-pages

tomdan-ai committed:
Fix erros found on doctors page

tomdan-ai committed:
Update file names

CaptainBawa committed:
Import sign-in and login component

CaptainBawa committed:
Add login auth

CaptainBawa committed:
Add sign-in auth

CaptainBawa committed:
Add styles for the login and sign-in components

CaptainBawa committed:
Remove routes on the DoctorList component

CaptainBawa committed:
Update routes

CaptainBawa committed:
Delete component

CaptainBawa committed:
Remove comment

CaptainBawa committed:
Remove comment

CaptainBawa committed:
Remove comment

CaptainBawa committed:
Remove comment

CaptainBawa committed:
Add margin-top

tomdan-ai committed:
Correct Doctor routes error

tomdan-ai committed:
Merge pull request #5 from CaptainBawa/doctor-pages-repair

Doctor pages repair

ridabensalem committed:
Add doctors components

ridabensalem committed:
import doctor component into app

ridabensalem committed:
Add missing field for adding doctor form

ridabensalem committed:
Add delete doctor functionality

ridabensalem committed:
Import delete doctor component

ridabensalem committed:
Update doctor list

ridabensalem committed:
Fix eslint errors

tomdan-ai committed:
Install necessary dependencies

tomdan-ai committed:
Make each doctor cards clickable link

tomdan-ai committed:
Create and implement doctor's profile page

tomdan-ai committed:
Create and implement user unique appointments page

tomdan-ai committed:
Add appointments and doctor detail's routes

tomdan-ai committed:
Fix Linter errors

ridabensalem committed:
Merge pull request #6 from CaptainBawa/addDoctor

Implement Add  delete doctor  functionality

CaptainBawa committed:
Change href

tomdan-ai committed:
Correct typo error

tomdan-ai committed:
Merge branch 'development' into doctor-details

tomdan-ai committed:
Merge pull request #7 from CaptainBawa/doctor-details

DocBook: Doctor details

tomdan-ai committed:
Fix linters caused by conflicts

Christelle-12 committed:
Add the Book -appointment component

Christelle-12 committed:
Fix add doctors selection

tomdan-ai committed:
Merge branch 'development' into book-appointment

CaptainBawa committed:
Merge pull request #9 from CaptainBawa/improve-auth-ui

Improve auth UI

tomdan-ai committed:
Add image assets to project

tomdan-ai committed:
Make delete request to appointment API

tomdan-ai committed:
Design Appointment components based on design guidelines

tomdan-ai committed:
Design Doctor's Details components based on design guidelines

tomdan-ai committed:
Add none Tailwind CSS

tomdan-ai committed:
Correct Linter errors in past files

Christelle-12 committed:
Fix userId and linters errors

Christelle-12 committed:
Merge remote-tracking branch 'origin/book-appointment' into book-appointment

Christelle-12 committed:
Fix linters

Christelle-12 committed:
Fix linters

CaptainBawa committed:
Add new div

CaptainBawa committed:
Add new div

CaptainBawa committed:
Add background image

tomdan-ai committed:
Merge pull request #10 from CaptainBawa/doc-details-style

DocBook: Doctor Details-design

CaptainBawa committed:
Merge pull request #8 from CaptainBawa/book-appointment

Book appointment

Christelle-12 committed:
Add responsiveness to add doctor page

Christelle-12 committed:
Add responsiveness to add doctor page

tomdan-ai committed:
Add Appointment location to appointments component

Christelle-12 committed:
Add back button functionalty to add doctor

Christelle-12 committed:
Add back button functionalty to add doctor

ridabensalem committed:
Install extra libraries

ridabensalem committed:
Add slider

ridabensalem committed:
Import doctorlist.css

ridabensalem committed:
Improve the UI/UX design

ridabensalem committed:
Refactor the doctorlist

ridabensalem committed:
Create Doctor list styles

ridabensalem committed:
Fix linters

ridabensalem committed:
Remove status and errors selectors

ridabensalem committed:
Add minum styles to the doctor list

ridabensalem committed:
Add alt to images

ridabensalem committed:
Center the doctor card

Add gaps

intialize the responsive classes

ridabensalem committed:
Install react icons

ridabensalem committed:
Add minum styles

ridabensalem committed:
Toggle navbar functionality

ridabensalem committed:
refactor doctor details

ridabensalem committed:
Make the circles bigger

ridabensalem committed:
Improve the UI

Christelle-12 committed:
Add styles for add appointment page

Christelle-12 committed:
Add book appointment navigation

Christelle-12 committed:
Fix linters

Christelle-12 committed:
Add hamburger functionality

Christelle-12 committed:
Add stylings for hamburger anf fix linters

Christelle-12 committed:
Improve close design

Christelle-12 committed:
Merge pull request #11 from CaptainBawa/styling

Add responsiveness to app

ridabensalem committed:
make the doctor list responsive

ridabensalem committed:
fix navbar issue

ridabensalem committed:
ADD position absolute

ridabensalem committed:
Add details's doctor

ridabensalem committed:
Fix linters

ridabensalem committed:
Add message alert

ridabensalem committed:
resolve conflicts

ridabensalem committed:
restore alert functionality

ridabensalem committed:
remove the hamburger from the home page

ridabensalem committed:
Merge pull request #12 from CaptainBawa/UI/UX

UI/UX

Christelle-12 committed:
Removed unnecessary styles

Christelle-12 committed:
Add back button to the page

Christelle-12 committed:
Deleted the unnecessary div

Christelle-12 committed:
Add responsiveness

Christelle-12 committed:
increased container's width

Christelle-12 committed:
Adjusted stylings

Christelle-12 committed:
Fix Linters

Christelle-12 committed:
Removed comments

Christelle-12 committed:
Fix style linters

Christelle-12 committed:
Merge pull request #13 from CaptainBawa/bugs

Bugs

CaptainBawa committed:
Merge branch 'development' into logout-func

CaptainBawa committed:
Merge pull request #14 from CaptainBawa/logout-func

DocBook. Milestone: Logout Functionality

tomdan-ai committed:
Update README.md

tomdan-ai committed:
Acknowledge behance designer in  README.md

tomdan-ai committed:
Add correct License

CaptainBawa committed:
Import logout component

CaptainBawa committed:
Add logout component

tomdan-ai committed:
Add live demo link to readme

tomdan-ai committed:
Update README.md

Meltrust commented on src/components/AddDoctor.jsx:
```diff
@@ -0,0 +1,190 @@
+import { faCircleArrowLeft } from '@fortawesome/free-solid-svg-icons';
+import { createDoctor } from '../redux/doctorsSlice';
+
+function AddDoctor() {
```
- [ ]  Please, kindly refactor this and all other functions in your project to ES6 arrow functions, so your code complies with the latest standards.

Refactoring is easy:

```javascript
const AddDoctor= () => {
  // ...
}
```

## Explanation

The `function` keyword has been soft deprecated in almost all the software industry in favor of ES6 arrow functions. This means, almost nobody uses it anymore, so much that it's considered a non-best practice.

You can do a search in Vscode for the "function" word, and correct as needed.  Shouldn't take you more than 5-10 min so don't worry (I found 8):

![image](https://github.com/CaptainBawa/DocBook-Front-End/assets/57421823/397bfaae-0195-4feb-8db1-8306598c03ae)


That way, your code will meet the latest standards. 💪

Meltrust commented on src/index.css:
```diff
@@ -0,0 +1,71 @@
+:root {
```
- [ ] In the desktop view, the details page does not look like the design given:


### Your design:

![image](https://github.com/CaptainBawa/DocBook-Front-End/assets/57421823/43eec46e-9cb1-4078-8f79-b3c84cb7a359)


### The requested design:

![image](https://github.com/CaptainBawa/DocBook-Front-End/assets/57421823/c5501f9a-78a7-42d2-9e0d-b3e9a27f996b)

Remember:  You need to be able to implement business requirements into a software solution.  This is because, in a job, all software solutions adhering to the given designs will be required

Therefore, kindly make it so your implementation looks a little more like the given design. 💪

Meltrust replied:
- [ ] In the desktop view, there is some inconsistency with the navigation, and the colors:

Add doctor has a green hamburger menu:

![image](https://github.com/CaptainBawa/DocBook-Front-End/assets/57421823/11c1d034-c398-4096-8c79-96eda296f4bf)

Add doctor has a green arrow icon:

![image](https://github.com/CaptainBawa/DocBook-Front-End/assets/57421823/fce1adf9-0e4a-4631-8e17-1491ba86ae89)

Delete Doctor has an arrow button, and it is blue:

![image](https://github.com/CaptainBawa/DocBook-Front-End/assets/57421823/4d7d2ef5-9194-4045-8392-f956c7b4b2e5)


Kindly, make it so the means of navigation is more consistent, and all the arrows/hamburgers have the same color to match the design. 👍

Meltrust replied:
- [ ] In the mobile view, this page is not rendering well because the content overflows the screen:

![image](https://github.com/CaptainBawa/DocBook-Front-End/assets/57421823/e190491c-4be9-43b5-8b7b-9e6e414b8044)

Kindly, add a little more styling so the info shows correctly. 👍

Meltrust commented on src/components/doctors/DoctorDetails.jsx:
```diff
@@ -0,0 +1,115 @@
+  createAppointment,
+} from '../../redux/appointmentsSlice';
+
+function DoctorDetails() {
```
- [ ] If I go to a doctors detail page, and press the "reserve appointment" button:

![image](https://github.com/CaptainBawa/DocBook-Front-End/assets/57421823/e140837c-b97b-42e0-97a2-cf69abb588d4)

It will take me to the reservations form, but it will make a request to the server trying to create the appointment before even taking me to the form.  Of course, that request will fail:

![image](https://github.com/CaptainBawa/DocBook-Front-End/assets/57421823/00cb4287-3df3-48bc-898f-acfc14cdc526)

Kindly, fix this bug so your app works as intended. 👍

Meltrust commented on src/components/appointments/Appointments.jsx:
```diff
@@ -0,0 +1,83 @@
+} from '../../redux/appointmentsSlice';
+import Navigation from '../Navigation';
+
+function Appointments() {
```
- [ ] It seems that all the users share the same appointments.  I tested this by creating multiple users and creating an appointment with one of them:

![image](https://github.com/CaptainBawa/DocBook-Front-End/assets/57421823/d60806cb-0779-4f4f-a156-82e7b4618e78)

![image](https://github.com/CaptainBawa/DocBook-Front-End/assets/57421823/52d79715-5bc6-4c96-ae62-9671b234334f)

Kindly, make it so each user can have their own appointments.  Tip: The backend work for this is already done and present in your backend project.

That way, your app will be more useful. 👍

Meltrust replied:
- [ ] You are including the patient's name in the "appointments" page, but every one of them is "John Doe":

![image](https://github.com/CaptainBawa/DocBook-Front-End/assets/57421823/905a049f-5a8c-403e-b98e-b352e317e929)

Kindly, make it so the user that made the appointment is shown their user name on that field. 👍

Meltrust reviewed (changes requested):
Hi @CaptainBawa ,

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation, but you are almost there!








#### You are really close to finishing the Microverse program!! Keep it up! 👍👍👍

![You can do it](https://media.giphy.com/media/FemTGpEZeK27M5fKL4/giphy.gif)

### After implementing the requested changes, please submit another review request. ♻️
_Check the comments under the review._







Cheers and Happy coding!👏👏👏

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the previous reviews unless it is requested otherwise._**

------

Christelle-12 committed:
Refactored to ES6 functions

Christelle-12 committed:
Changed functions to use ES6 ssyntax

CaptainBawa committed:
Merge pull request #17 from CaptainBawa/userid-doctorid-bugs

DocBook. UserId And doctorId Bug Fix

Christelle-12 committed:
Merge pull request #16 from CaptainBawa/ES6-functions

Es6 functions

ridabensalem committed:
Change the color of backbutton and header table

ridabensalem committed:
modify details doctor styles

ridabensalem committed:
Match the design of doctor details page

ridabensalem committed:
Merge pull request #18 from CaptainBawa/UI/UX

UI/Ux

tomdan-ai committed:
Add missing dependencies

tomdan-ai committed:
Fix reserve appointment button

tomdan-ai committed:
Fix Linter errors

CaptainBawa committed:
Find temporary fix for the user id bug

CaptainBawa committed:
Find temporary fix for the user id bug

CaptainBawa committed:
Remove commented code

ridabensalem committed:
Make the appointement responsive

CaptainBawa committed:
Merge pull request #19 from CaptainBawa/app-corrections

App corrections

ridabensalem committed:
Fix responsive issues

ridabensalem committed:
change the xs font-size

ridabensalem committed:
center the table on mobile version

ridabensalem committed:
Merge pull request #20 from CaptainBawa/UI/UX

UI/Ux

CaptainBawa commented:
### Dear Reviewer 

We've addressed all the requested changes from the previous reviewer; however, there's one bug we've encountered for which we've implemented a temporary fix. We are actively collaborating to identify a permanent solution.

The issue pertains to the uniqueness of the user associated with appointment creation. Previously, the patient's name in the appointment table was consistently displayed as "John Doe," regardless of the actual creator. To rectify this, we recognize the need to link the logged-in user's ID to the appointment before sending the request to the API. We've attempted to implement this but have encountered difficulties in doing so. As an interim measure, we've allowed users to manually select their usernames when booking an appointment to obtain the necessary ID.

Please be aware that this is a temporary workaround, and we are actively exploring avenues for a more permanent resolution. Unfortunately, time constraints are pressing, which is why we are providing this update to keep you informed of our ongoing efforts. Thank you for your understanding.

mcrd25 reviewed (approved):
Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Made requested changes 👍 
## Optional suggestions
- Don't forget that the application needs to look exactly like design with exception of photos, titles and descriptions
- There are some candidates to make some of these html components reusable.
- 
*  I am not seeing tests for your Front-End code - this is something we expect to be implemented for the final evaluation. Here are some resources you can use:
   *  https://reactjs.org/docs/testing.html
   * https://jestjs.io/docs/tutorial-react
   * https://www.freecodecamp.org/news/testing-react-hooks/
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

**Hope all is well, and that my comments were helpful. 😸**

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
_Also, you do not have to implement all requested changes, as this is up to you in the end, but I strongly recommend you implement them._

CaptainBawa committed:
Merge pull request #21 from CaptainBawa/redux-testing

DocBook. Milestone: Redux Testing

CaptainBawa committed:
Setup jest and react testing library

CaptainBawa committed:
Add test for the usersSlice

CaptainBawa committed:
Add test for the doctorsSlice

CaptainBawa committed:
Add test for the appointmentsSlice

CaptainBawa committed:
Delete comment

CaptainBawa committed:
Reconfigure linters

CaptainBawa committed:
Reconfigure linters

CaptainBawa committed:
Fix dependencies conflict
